packages/block-editor/src/components/use-setting/index.js

```diff
@@ -93,126 +94,158 @@ const removeCustomPrefixes = ( path ) => {
 	return prefixedFlags[ path ] || path;
 };
 
+const mergeCache = new WeakMap();
+function mergeOrigins( value ) {
+	let result = mergeCache.get( value );
+	if ( ! result ) {
+		result = [ 'default', 'theme', 'custom' ].flatMap(
+			( key ) => value[ key ] ?? []
+		);
+		mergeCache.set( value, result );
+	}
+	return result;
+}
+
 /**
- * Hook that retrieves the given setting for the block instance in use.
+ * Hook that retrieves the given settings for the block instance in use.
  *
  * It looks up the settings first in the block instance hierarchy.
- * If none is found, it'll look it up in the block editor store.
+ * If none are found, it'll look them up in the block editor settings.
  *
- * @param {string} path The path to the setting.
- * @return {any} Returns the value defined for the setting.
- * @example
- * ```js
- * const isEnabled = useSetting( 'typography.dropCap' );
- * ```
+ * @param {string[]} paths The paths to the settings.
+ * @return {any[]} Returns the values defined for the settings.
  */
-export default function useSetting( path ) {
-	const { name: blockName, clientId } = useBlockEditContext();
+export function useSettings( paths ) {
```

Comment:
did you consider using variable argument counts here? it seems like most cases we'd be calling this we'll have literal values vs. dynamically-created arrays and I wonder if that's a convenience that would be nice.

```js
const [ … ] = useSettings( 'setting-a', 'setting-b', 'color.setting.thing' );
```

Reply:
Yes, that's a good idea. Variable arguments are better than an array argument. This is quite related to @tyxla's feedback about the module name:

> Does it make sense to move `useSettings()` to a separate file? Perhaps under `../use-settings/index.js`? Right now it's a bit unfortunate that `use-setting/index.js` exports `useSettings()`.

To solve both problems, we could do this:
1. Rewrite `useSettings` to use variable arguments.
2. Migrate all remaining usages of `useSetting` to `useSettings`. The entire difference is the return value as array: `const [ a ] = useSettings( 'a' )` vs `const a = useSetting( 'a' )`.
3. This makes `useSetting` practically deprecated: there's no good reason any more to use it instead of `useSettings`.
4. The actual source file can then be renamed to `use-settings/index.js`. The main thing it exports is `useSettings`, and then there's also the little `useSetting` back-compat wrapper. I think that solves the naming concerns very well.

Thoughts?

Reply:
I really like that idea 👍 one fewer new API while still supporting all original use cases. Any good reason not to go that way?

Reply:
Implemented! All usages of `useSetting` are refactored to `useSettings`, and the source folder name is now `use-settings`.